### PR TITLE
Migrate cloud resource manager v2 to v3 API

### PR DESCRIPTION
**Pull request by iyabchen**
Migrate cloud resource manager v2 to v3 API. 
This is to support terraform validator to use v3 API to fetch project and folder parents to construct the list of ancestors.  

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Review by melinath — COMMENTED** [review]
It looks like resourceManagerV2 is relatively rarely used - would it be possible to upgrade v2 usage to v3 across the board rather than adding a third version of this library?

If this is trivial to do, it would be neat, but I don't think it's a blocker if it looks like it would be complicated.

**Comment by iyabchen** [comment]
> It looks like resourceManagerV2 is relatively rarely used - would it be possible to upgrade v2 usage to v3 across the board rather than adding a third version of this library?
> 
> If this is trivial to do, it would be neat, but I don't think it's a blocker if it looks like it would be complicated.

Is there any place/tests to check whether V2 is being referenced by other libraries?

**Comment by melinath** [comment]
It looks like v2 is only used in 7 files: https://github.com/GoogleCloudPlatform/magic-modules/search?q=resourceManagerV2

v1 is used in a number of places and is (I think) more different, so probably leaving it alone is better for now.

In terms of external dependencies - I don't think we need to worry about that. There aren't any other cases I'm aware of where people would be interacting with terraform at this level.

File: mmv1/third_party/terraform/utils/provider.go.erb [hunk 1]
```diff
@@ -168,6 +168,7 @@ func Provider() *schema.Provider {
 			DataflowCustomEndpointEntryKey:               DataflowCustomEndpointEntry,
 			IamCredentialsCustomEndpointEntryKey:         IamCredentialsCustomEndpointEntry,
 			ResourceManagerV2CustomEndpointEntryKey:      ResourceManagerV2CustomEndpointEntry,
```
Comment (melinath) [inline mmv1/third_party/terraform/utils/provider.go.erb:170]: It should be possible to remove this line as well.

File: mmv1/third_party/terraform/utils/provider_handwritten_endpoint.go.erb [hunk 2]
```diff
@@ -79,6 +79,16 @@ var ResourceManagerV2CustomEndpointEntry = &schema.Schema{
 	}, DefaultBasePaths[ResourceManagerV2BasePathKey]),
 }
```
Comment (melinath) [inline mmv1/third_party/terraform/utils/provider_handwritten_endpoint.go.erb:80]: It should be possible to remove the above section.

**Review by melinath — COMMENTED** [review]
It looks like there are still some trailing references to `ResourceManagerV2` with various capitalizations that could be removed.

Since a v2 custom endpoint can be set by users, it may be a breaking change to switch to an uncustomized v3 endpoint. Falling back to v2 to maintain current use cases may be sufficient but I'm not sure - I'll look into it. I don't remember how frequently exactly people are expected to use those endpoint variables; they may just be for testing.

**Comment by slevenick** [comment]
Looks like only Folder tests are failing, and those are unrelated. I'd prefer not to merge until we fix that though.

cc: @ScottSuarez who is assigned on https://github.com/hashicorp/terraform-provider-google/issues/10286

**Comment by ScottSuarez** [comment]
Folder tests are failing due to what looks like an internal latency between folder creation and a folder being searchable.
created b/94275557

@slevenick to decide if the pr can continue while API team addresses our blocker.

**Comment by melinath** [comment]
I think this is blocked on https://github.com/GoogleCloudPlatform/magic-modules/pull/5989

**Comment by melinath** [comment]
@iyabchen this should be unblocked now - once you resolve the conflicts it should rerun the tests for you.

**Review by melinath — APPROVED** [review]
LGTM - the remaining failure is also happening on TeamCity nightly builds: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaGoogleProject/288007?buildTab=tests&name=TestAccCloudAssetFolderFeed_cloudAssetFolderFeedExample&pager.currentPage=1&expandedTest=7193717083169324042

@slevenick any objections to moving forward?

**Comment by slevenick** [comment]
Looks alright to me!

**Pull request merged**